core/varianthandler.cpp

```diff
@@ -322,6 +323,15 @@ QString VariantHandler::displayString(const QVariant &value)
         return (*it.value())(value);
     }
 
+    if (value.canConvert<QJsonObject>()) {
```

Comment:
This probably shouldn't be canConvert() but userType() == qMetaTypeId<QJsonObject>(), +2 otherwise.